contracts/contract_state_tree.md

```diff
@@ -62,3 +62,7 @@ the latest block only contains the calls in that block.
 The key is `<contract_address><caller_address><caller_nonce><contract_address>`.
 
 See [Call Serialization](../serializations.md#contract-call) for the serialization specification.
+
+Each call includes the caller address and caller nonce that, together,
+correlate the call with the transaction in the block that did the
```

Comment:
This comes from investigation of https://www.pivotaltracker.com/story/show/154664693
for which I am considering to determine the gas fees to give to the miner
by exploiting the `gas_used` already in each contract call object in the state tree of the block.
But, as the call object does not include the gas price, I am checking if possible to correlate the call object to the create tx or call tx in the same block in order to retrieve the gas price from that. And it seems possible correlating on caller address and caller nonce.

Other option would be to duplicate the gas price in the call object.

Other option is attempting to identify further alternatives.

Reply:
> I am checking if possible to correlate the call object to the create tx or call tx in the same block in order to retrieve the gas price from that. And it seems possible correlating on caller address and caller nonce.

It seems [a contract can call another contract](https://github.com/aeternity/epoch/blob/c8050fea2b033b9448d1e907befcaccfa17b24ac/apps/aecore/src/aec_vm_chain.erl#L113) that seems to create call object with caller address and nonce of the contract (not double checked). Also maybe [a contract may create another contract](https://github.com/aeternity/epoch/blob/c8050fea2b033b9448d1e907befcaccfa17b24ac/apps/aevm/src/aevm_eeevm.erl#L1458) (via `aect_create_tx`?) potentially creating call object (not double checked). Hence I am not sure (twice) it is possible to correlate call object to create or call tx.

Reply:
Call objects are only created for top-level (tx) calls. The top level call will include all gas used by any recursive calls.